### PR TITLE
Fix yaml merging when an override is defined in the same file as the source.

### LINKED ISSUE #14060: playtest: "Invalid <Option>" when loading this map in multiplayer
Hey,
made[ this map](http://resource.openra.net/maps/22831/) which works fine in playtest20170902 skirmish. No playtest Lint errors. (some random warnings)
In multiplayer however I can't choose techlevel or AI, even though that is perfectly possible in skirmish mode. 
EDIT: I can choose low or medium tech, but no idea why. I reconfigured all techlevels after all.

21:15:20 - pchote: interesting
21:15:27 - pchote: seems to be related to the resource center fetching
...
21:33:58 - pchote: afaik none of the code related to this bug has changed recently
...
21:34:57 - pchote: the key point is that this only happens when a dedicated server doesn't have the map installed

### PR DESCRIPTION
**Pull request by pchote**
Fixes #14060.

### CODE REVIEW

File: OpenRA.Game/MiniYaml.cs [hunk 1]
```diff
@@ -325,6 +327,42 @@ static List<MiniYamlNode> ResolveInherits(string key, MiniYaml node, Dictionary<
 			return resolved;
 		}
 
+		/// <summary>
+		/// Merges any duplicate keys are are defined within the same set of nodes.
```
Comment (reaperrr) [inline OpenRA.Game/MiniYaml.cs]: "that are" instead of "are are".

Comment (pchote) [reply OpenRA.Game/MiniYaml.cs]: Fixed.

File: OpenRA.Test/OpenRA.Game/MiniYamlTest.cs [hunk 2]
```diff
@@ -169,5 +169,30 @@ public void EmptyLinesShouldCountTowardLineNumbers()
...
+	Merge: override
+		Child: override
+	Override:
+";
+
+			var result = MiniYaml.Merge(new[] { baseYaml }.Select(s => MiniYaml.FromString(s, "")));
+			Assert.IsTrue(result.Count(n => n.Key == "Test") == 1, "Result should have exactly one Test node, but does not.");
+
+			var testNodes = result.First(n => n.Key == "Test").Value.Nodes;
+			Assert.IsTrue(testNodes.Count == 3, "Merged Test node should have exactly three children, but has {0}.".F(testNodes.Count));
```
Comment (RoosterDragon) [inline OpenRA.Test/OpenRA.Game/MiniYamlTest.cs]: Consider `Assert.That(testNodes.Select(n => n.Key), Is.EqualTo(new[] { "Merge", "Original", "Override" }), "Merged Test node has incorrect child nodes.");` to assert the names and ordering are correct as well as the count.

File: OpenRA.Test/OpenRA.Game/MiniYamlTest.cs [hunk 3]
```diff
@@ -169,5 +169,30 @@ public void EmptyLinesShouldCountTowardLineNumbers()
...
+
+			var mergeNode = testNodes.First(n => n.Key == "Merge").Value;
+			Assert.True(mergeNode.Value == "override", "Merge node value should be 'override', but is not");
```
Comment (RoosterDragon) [inline OpenRA.Test/OpenRA.Game/MiniYamlTest.cs]: Consider `Assert.That(mergeNode.Value, Is.EqualTo("override"), "Merge node has incorrect value.");`. NUnit then spits out the expected and actual values for you, saving a bit of boilerplate in the messages.

**Review by RoosterDragon — DISMISSED** [review]
:+1: Tests make things so much easier to fathom - I love tests. :)

File: OpenRA.Test/OpenRA.Game/MiniYamlTest.cs (hunk 2 above)
Comment (pchote) [reply OpenRA.Test/OpenRA.Game/MiniYamlTest.cs]: > Expected is <System.String[3]>, actual is <System.Linq.Enumerable+SelectListIterator`2[OpenRA.MiniYamlNode,System.String]>

:grin:

**Comment by pchote** [comment]
Updated.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b5e554775dc3dd05147be39770dd9b89f22cb05d...1781eb44c963d8e367e9d62c1b8e182e7ded02dd

**Pull request merged**